test/ScopeExitTest.cpp

```diff
@@ -0,0 +1,77 @@
+#include "pch.h"
+#include <kf/ScopeExit.h>
+#include <kf/USimpleString.h>
+
+static int g_functionCallCount = 0;
+
+void testFunction()
+{
+    ++g_functionCallCount;
+}
+
+SCENARIO("kf::ScopeExit")
+{
+    GIVEN("A ScopeExit with a valid function")
+    {
+        g_functionCallCount = 0;
+
+        WHEN("The ScopeExit object is created")
+        {
+            {
+                auto scopeExit = kf::MakeScopeExit() += [] { testFunction(); };
```

Comment:
Don't use kf::MakeScopeExit directly, it's supposed to be always used via a macro.